packages/plugin-ext/src/plugin/scm.ts

```diff
@@ -511,6 +515,22 @@ class SsmResourceGroupImpl implements theia.SourceControlResourceGroup {
         return rawResourceSplices;
     }
 
+    protected getThemeIcons(icons: (UriComponents | ThemeIcon | undefined)[]): (UriComponents | ThemeIcon | undefined)[] {
+        const processedIcons: (UriComponents | ThemeIcon | undefined)[] = [];
+        for (const icon of icons) {
+            processedIcons.push(this.processIcon(icon));
+        }
+        return processedIcons;
+    }
+
+    protected processIcon(icon: UriComponents | ThemeIcon | undefined): UriComponents | ThemeIcon | undefined {
```

Comment:
We probably need a better name than `processIcon`, what is it attempting to do?

Reply:
Yeah,
I could not think of a better way to describe what is going on here.
The function will essentially do nothing if the input is `undefined` or a `ThemeIcon`.
Then it will change the path of the `URI` to the relative path needed for displaying through the server host instead of the file path to reach the icon. It is not the complete path as that is taken care of in `scm-main.ts` where you actually put the port number needed. So I am not sure what to call it.